### ISSUE TITLE
auto-rerun github workflow failures

### ISSUE THREAD

**Issue by moul** [gnolang/gno#2825]
https://github.com/orgs/community/discussions/67654#discussioncomment-8038649

Could be nice for codecov.

**Comment by thehowl** [comment]
I don't believe we should do this, particularly about codecov; we should disable status checks for codecov so that it is only informational and doesn't make the CI red (it is systematically ignored anyway at present).

There are other tests that are flaky and might benefit from being re-run; however I fear that auto-rerunning them on failure will hide away the fact that those tests are still flaky and failing. Like putting a device that automatically adds engine coolant to your car whenever it drops below the minimum: definitely cool, but ignores the problem that the car shouldn't be leaking it in the first place.

**Comment by moul** [comment]
Codecov has 2 issues; one of them is because it's misconfigured. Here we can easily do what you said. The other problem is that sometimes we just have API errors at upload (HTTP422 iirc). For this case and similar ones I suggest having a retry.
